civis/io/_files.py

```diff
@@ -132,25 +134,21 @@ def _multipart_upload(buf, name, file_size, client, **kwargs):
     assert num_parts == len(urls), \
         "There are {} file parts but only {} urls".format(num_parts, len(urls))
 
-    # generate for writing a specific number of bytes from the buffer
-    def _gen_chunks(part_buf, max_bytes, chunk_size=32 * 1024):
-        bytes_read = 0
-        while True:
-            length = min(chunk_size, max_bytes - bytes_read)
-            if length <= 0:
-                break
-            bytes_read += length
-            data = part_buf.read(length)
-            yield data
-
     # upload function wrapped with a retry decorator
     @retry(RETRY_EXCEPTIONS)
-    def _upload_part_base(item, part_path):
+    def _upload_part_base(item, buf, part_size, file_size):
         part_num, part_url = item[0], item[1]
+        offset = part_size * part_num
+        num_bytes = min(part_size, file_size - offset)
+
         log.debug('Uploading file part %s', part_num)
-        file_out = part_path.format(part_num)
-        with open(file_out, 'rb') as fout:
-            part_response = requests.put(part_url, data=fout)
+        mode = 'r'
+        if 'b' in buf.mode:
+            mode += 'b'
+        with open(buf.name, mode) as fout:
```

Comment:
I find it confusing to name a read-only buffer `fout`. I use that name for files I'm writing. Maybe that's just me? ... Oh, I see you have naming conventions reversed from what I use. It looks like this repo currently uses both conventions.

Reply:
I agree! I think your convention makes more sense. I've updated references to use `fin` for read and `fout` for write.